rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -1570,15 +1571,48 @@ private void adjustBalancesIfChangeOutputWasDust(BtcTransaction btcTx, Coin sent
      * The hash for the signature must be calculated with Transaction.SigHash.ALL and anyoneCanPay=false. The signature must be canonical.
      * If enough signatures were added, ask federators to broadcast the btc release tx.
      *
-     * @param federatorPublicKey   Federator who is signing
+     * @param federatorBtcPublicKey   Federator who is signing
      * @param signatures           1 signature per btc tx input
-     * @param rskTxHash            The id of the rsk tx
+     * @param rskTxHashSerialized            The id of the rsk tx
      */
-    public void addSignature(BtcECKey federatorPublicKey, List<byte[]> signatures, byte[] rskTxHash) throws Exception {
+    public void addSignature(BtcECKey federatorBtcPublicKey, List<byte[]> signatures, byte[] rskTxHashSerialized) throws Exception {
+        if (signatures == null || signatures.isEmpty()) {
+            return;
+        }
+
         Context.propagate(btcContext);
 
+        Keccak256 rskTxHash = new Keccak256(rskTxHashSerialized);
```

Comment:
nope, it throws an `IllegalArgumentException`. See the [implementation](https://github.com/rsksmart/rskj/blob/refactor_add_signature/rskj-core/src/main/java/co/rsk/crypto/Keccak256.java#L45).

Reply:
you are right actually, the hash is always valid.
Regarding the keccak suggestion, I think is just a matter of taste, because in the BridgeSupport implementation we are using both the keccak hash and the serialized hash

Reply:
I think it's cleaner to receive the Keccack object, it ensures you will be working with a valid element. Receiving an array of bytes, could be anything. That would force you to do some validations since you can't be sure that the value was previously validated

Reply:
make sense! I'll update it

Reply:
this was updated